osu.Game/Screens/Play/BreakOverlay.cs

```diff
@@ -96,60 +104,78 @@ public BreakOverlay(bool letterboxing)
                         Anchor = Anchor.Centre,
                         Origin = Anchor.Centre,
                     }
-                }
-            };
-        }
-
-        protected override void LoadComplete()
-        {
-            base.LoadComplete();
-            initializeBreaks();
-        }
-
-        private void initializeBreaks()
-        {
-            if (!IsLoaded) return; // we need a clock.
-
-            FinishTransforms(true);
-            Scheduler.CancelDelayedTasks();
-
-            if (breaks == null) return; //we need breaks.
-
-            foreach (var b in breaks)
-            {
-                if (!b.HasEffect)
-                    continue;
+          }
+      };
+    }
 
-                using (BeginAbsoluteSequence(b.StartTime, true))
-                {
-                    fadeContainer.FadeIn(fade_duration);
-                    breakArrows.Show(fade_duration);
+    protected override void LoadComplete()
+    {
+      base.LoadComplete();
+      initializeBreaks();
+    }
+    [BackgroundDependencyLoader]
+    private void load(OsuConfigManager config)
+    {
+      // We need BackgroundOpacityOffset to lighten during breaks.
+      if (BackgroundOpacityOffset == null) return;
+
+      lightenDuringBreaks = config.GetBindable<bool>(OsuSetting.LightenDuringBreaks);
+      lightenDuringBreaks.ValueChanged += _ => breakLighten.TriggerChange();
```

Comment:
Take below action and invoke it directly here. There is no need to use one event to trigger the next one and then execute something.

Reply:
I am also confused here for reasons stated above.

Reply:
I could do something like this if that's what you meant:
```c#
[BackgroundDependencyLoader]
private void load(OsuConfigManager config)
{
    // We need BackgroundOpacityOffset to lighten during breaks.
    if (BackgroundOpacityOffset == null) return;

    lightenDuringBreaks = config.GetBindable<bool>(OsuSetting.LightenDuringBreaks);
    lightenDuringBreaks.ValueChanged += _ => updateBackgroundOpacityOffset();
    currentlyOnBreak.ValueChanged += _ => updateBackgroundOpacityOffset();
}
private void updateBackgroundOpacityOffset()
{
    this.TransformBindableTo(
    BackgroundOpacityOffset,
    lightenDuringBreaks.Value && currentlyOnBreak.Value ? 1 : 0,
    fade_duration,
    Easing.OutQuint
    );
}
```

Reply:
Something like that yeah. Not sure if you need a bindable for that as you could have a simple bool to do the same thing.

Reply:
How so?

Reply:
Oh. No that comment was made on a misunderstanding. Don't mind it now.